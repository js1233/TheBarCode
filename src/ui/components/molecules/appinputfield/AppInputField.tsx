import React, { useState } from "react";
import {
  StyleProp,
  StyleSheet,
  TextInput,
  TextInputProps,
  TextStyle,
  View,
  ViewStyle
} from "react-native";
import { COLORS, FONTS, FONT_SIZE, SPACE } from "config";
import { ImageStyle } from "react-native";
import { useEffect } from "react";
import { usePreferredTheme } from "hooks";

export interface AppInputFieldProps extends TextInputProps {
  style?: StyleProp<TextStyle>;
  // default undefined
  valueToShowAtStart?: string;
  // default false
  shouldDisable?: boolean;
  secureTextEntry?: boolean;

  rightIcon?: () => React.ReactElement | null;
  leftIcon?: () => React.ReactElement | null;
  viewStyle?: StyleProp<ViewStyle>;
  iconStyle?: StyleProp<ImageStyle>;
  multiline?: boolean;
  textAlignVertical?: any;
  onClickedOnRightIcon?: () => void;
}

type Props = AppInputFieldProps;

export const AppInputField = React.memo<Props>(
  ({
    onChangeText,
    style,
    valueToShowAtStart,
    shouldDisable = false,
    rightIcon,
    leftIcon,
    viewStyle,
    //iconStyle,
    multiline = false,
    textAlignVertical,
    secureTextEntry = false,
    //onClickedOnRightIcon,
    //iconStyle,
    ...rest
  }) => {
    const [value, setTextInputValue] = useState(valueToShowAtStart);
    const { themedColors } = usePreferredTheme();

    useEffect(() => {
      setTextInputValue(valueToShowAtStart);
    }, [valueToShowAtStart]);

    const getMultiline = () => (multiline ? styles.multiline : {});

    return (
      <View style={[styles.input, viewStyle]}>
        {leftIcon && (
          <View style={styles.leftIconView}>
            {leftIcon ? leftIcon() : null}
          </View>
        )}
        <TextInput
          textAlignVertical={
            textAlignVertical ? textAlignVertical : "center"
          }
          testID="InputField"
          value={value}
          secureTextEntry={secureTextEntry}
          editable={!shouldDisable}
          onChangeText={(text) => {
            onChangeText?.(text);
            setTextInputValue(text);
          }}
          placeholderTextColor={themedColors.primaryShade["900"]}
          style={[
            styles.textInput,
            style,
            getMultiline(),
            shouldDisable
              ? { backgroundColor: COLORS.theme?.interface[200] }
              : null
          ]}
          multiline={multiline}
          {...rest}
        />
        {rightIcon && (
          // <TouchableOpacity
          //   onPress={onClickedOnRightIcon}
          //   style={{ alignItems: "center", justifyContent: "center" }}>
          <View style={styles.rightIconView}>
            {rightIcon ? rightIcon() : null}
          </View>
          //   </TouchableOpacity>
        )}
      </View>
    );
  }
);

const styles = StyleSheet.create({
  input: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "stretch",
    color: COLORS.black,
    borderStyle: "solid",
    borderRadius: 6,
    flex: 1
  },
  multiline: {
    paddingBottom: SPACE.sm
  },
  textInput: {
    fontFamily: FONTS.regular,
    fontSize: FONT_SIZE.xs,
    flex: 1,
    color: COLORS.black,
    textAlignVertical: "center",
    paddingHorizontal: SPACE.md
  },
  leftIconView: {
    alignItems: "center",
    justifyContent: "center",
    paddingStart: SPACE.md
  },
  rightIconView: {
    alignItems: "center",
    justifyContent: "center",
    alignSelf: "center",
    paddingEnd: SPACE.md
  }
});
