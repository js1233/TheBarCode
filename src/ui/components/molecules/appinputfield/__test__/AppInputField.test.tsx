import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { AppInputField } from "ui/components/molecules/appinputfield/AppInputField";

test("snapshot testing", () => {
  const rendered = render(<AppInputField />).toJSON();
  expect(rendered).toMatchSnapshot();
});

describe("AppInputField", () => {
  //fireEvent.changeText(getByTestId("InputField"), "Hello world");

  const inputText = "Hello World";
  let getByTestId: any;

  beforeEach(() => {
    ({ getByTestId } = render(
      <AppInputField valueToShowAtStart={inputText} />
    ));
  });

  it("test the on change text", function () {
    fireEvent.changeText(getByTestId("InputField"), inputText);
  });

  it("test the value of input field", function () {
    expect(getByTestId("InputField").props.value).toEqual(inputText);
  });

  it("clears the input text", function () {
    fireEvent.changeText(getByTestId("InputField"), "");
    expect(getByTestId("InputField").props.value).toEqual("");
  });
});
