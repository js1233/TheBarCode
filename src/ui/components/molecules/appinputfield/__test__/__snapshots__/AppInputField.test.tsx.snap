// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "borderRadius": 6,
        "borderStyle": "solid",
        "color": "#000000",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <TextInput
    editable={true}
    multiline={false}
    onChangeText={[Function]}
    placeholderTextColor="#134E4A"
    secureTextEntry={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "flex": 1,
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 25,
          "paddingHorizontal": 12,
          "textAlignVertical": "center",
        },
        undefined,
        Object {},
        null,
      ]
    }
    testID="InputField"
    textAlignVertical="center"
  />
</View>
`;
